pkg/compose/hash.go

```diff
@@ -37,5 +38,22 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	for _, serviceConfig := range o.Configs {
+		if projectConfig, ok := project.Configs[serviceConfig.Source]; ok {
```

Comment:
nit: I would invert the if statements so that we do not get the code so nested

Reply:
@jhrotko done